File: #0513.find-bottom-left-tree-value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        
        pair<int,int> result = {0, root->val};
        findBottomLeftValue(root, 0, result);
        return result.second;
    }

private:
    void findBottomLeftValue(TreeNode* root, int level,
        pair<int,int>& result) {

        if (!root) {
            return;
        }

        if (level>result.first) {
            result = {level, root->val};
        }

        findBottomLeftValue(root->left, level + 1, result);
        findBottomLeftValue(root->right, level + 1, result);
    }
};

File: #0513.find-bottom-left-tree-value.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findBottomLeftValue = function(root) {
    let arr = [];
    findBottomLeftValue(root, 1);
    return arr[arr.length - 1];

    function findBottomLeftValue(node, level) {
        if (node === null) {
            return;
        }
        
        findBottomLeftValue(node.right, level + 1);
        arr[level] = node.val; 
        findBottomLeftValue(node.left, level + 1);
    }
};
